samtranslator/model/stepfunctions/events.py

```diff
@@ -135,8 +144,12 @@ def to_cloudformation(self, resource, **kwargs):  # type: ignore[no-untyped-def]
         events_rule.Name = self.Name
         events_rule.Description = self.Description
 
-        role = self._construct_role(resource, permissions_boundary)  # type: ignore[no-untyped-call]
-        resources.append(role)
+        role: Union[IAMRole, str]
```

Comment:
I think you need to add `Dict` here as well. If the user puts intrinsics then the type will be `Dict[str, Any]`